src/core/server/plugins/plugins_service.ts

```diff
@@ -86,6 +86,9 @@ export interface PluginsServiceDiscoverDeps {
   environment: InternalEnvironmentServicePreboot;
 }
 
+const OSS_PATH_REGEX = /[\/|\\]src[\/|\\]plugins[\/|\\]/; // Matches src/plugins directory on POSIX and Windows
+const XPACK_PATH_REGEX = /[\/|\\]x-pack[\/|\\]plugins[\/|\\]/; // Matches x-pack/plugins directory on POSIX and Windows
+
```

Comment:
I was thinking it'd be nice if we added a `source` property to the `PluginWrapper` class inside `src/core/server/plugins/plugin.ts` and populated it in the constructor. Then we could use this information in other places as well rather than only inside the service.

I'm fairly confident that it would be a small change to this PR, WDYT?

Reply:
Just to clarify: the `PluginWrapper` class already has a `path` field.

Do you envision the `source` field as a string literal such as `'x-pack' | `'oss'`?